pip/utils/outdated.py

```diff
@@ -135,12 +135,14 @@ def pip_version_check(session):
 
         # Determine if our pypi_version is older
         if pip_version < pkg_resources.parse_version(pypi_version):
-            logger.warning(
-                "You are using pip version %s, however version %s is "
-                "available.\nYou should consider upgrading via the "
-                "'pip install --upgrade pip' command." % (pip.__version__,
-                                                          pypi_version)
-            )
+            # Check to make sure we reminding someone already upgrading
+            if " ".join(sys.argv) == "pip install --upgrade pip":
```

Comment:
This is very fragile. Won't work if people use ```python -m pip``` (necessary to upgrade pip on Windows) or if they use -U. Or for a lot of other possible reasons. I like the idea, but can you check via some other means (e.g., the command is "install", the "upgrade" option is set and pip is in the list of requirements?